### PR TITLE
Bug 1957041: Update e2echart to include node roles info

### PR DESCRIPTION
**Pull request by ihcsim**
This PR updates the `node-state` section in the e2echart with node roles, by
embedding the information (in the form of `roles/$roles`) in the `Message` field. 
This change preserves the dependency some tests have on the event
messages, where certain prefix and suffix are expected, as seen in
[pkg/synthetictests/kubelet.go](https://github.com/openshift/origin/blob/36b43d1c8df3a74e1e1fca5e8a9c1c6824d67fe6/pkg/synthetictests/kubelet.go#L132-L134) and [pkg/monitor/intervalcreation/node.go](https://github.com/openshift/origin/blob/36b43d1c8df3a74e1e1fca5e8a9c1c6824d67fe6/pkg/monitor/intervalcreation/node.go#L44-L46)  

This is what the updated section looks like:
image  The `eventIntervals` JS object used for testing can be found in this [gist](https://gist.github.com/ihcsim/8e5b70883b5335de46a88046625e19ad). 

Signed-off-by: Ivan Sim <isim@redhat.com>

### CODE REVIEW

**Comment by deads2k** [comment]
Can you group the masters at the top of the section and the workers at the bottom of the section?

File: pkg/monitor/intervalcreation/node.go [hunk 1]
```diff
@@ -46,6 +55,15 @@ func IntervalsFromEvents_NodeChanges(events monitorapi.Intervals, beginning, end
 		}
 		reason := strings.SplitN(strings.TrimPrefix(event.Message, "reason/"), " ", 2)[0]
 
+		var roles string
```
Comment (deads2k) [inline pkg/monitor/intervalcreation/node.go:58]: factor into a function I can call kind of like https://github.com/deads2k/origin/blob/bbd9949c834a1491ac62a9b74ea3fdb4bb5f6840/pkg/monitor/monitorapi/identification_operator.go#L10-L11

**Comment by deads2k** [comment]
minor factorization request.  Looks good otherwise.

File: pkg/monitor/intervalcreation/node.go (hunk 1 above)
Comment (ihcsim) [reply pkg/monitor/intervalcreation/node.go]: updated. see 188c0fcfd11f19281d8c86d3bfeba0c27042754a.

**Comment by deads2k** [comment]
verify failure is real

**Comment by ihcsim** [comment]
@deads2k i fixed the `verify` errors. i don't know why `/retest` keeps running on an old commit that was rebased.

**Comment by deads2k** [comment]
This is waiting on having masters grouped next to each other in the node view.

**Comment by ihcsim** [comment]
@deads2k the roles grouping is fixed now.

image

**Comment by deads2k** [comment]
This is now missing the "not ready" phase.  https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.8-e2e-azure-upgrade/1387240645822779392

I like the sorting though.

**Comment by ihcsim** [comment]
@deads2k This should be fixed now. See CI result [here](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/26093/pull-ci-openshift-origin-master-e2e-gcp-upgrade/1387470897031417856). Thanks!

File: pkg/monitor/intervalcreation/node_test.go [hunk 2]
```diff
@@ -1,29 +0,0 @@
-package intervalcreation
-
-import (
-	"testing"
-	"time"
-
-	monitorserialization "github.com/openshift/origin/pkg/monitor/serialization"
-)
-
-func TestIntervalsFromEvents_NodeChanges(t *testing.T) {
```
Comment (deads2k) [inline pkg/monitor/intervalcreation/node_test.go:10]: What happened to this test?

Comment (ihcsim) [reply pkg/monitor/intervalcreation/node_test.go]: my bad... i meant to revert my changes in this file. i restored the file.

**Pull request merged**